calculator/src/main/java/edu/nextstep/camp/calculator/CalculatorViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package edu.nextstep.camp.calculator
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import edu.nextstep.camp.calculator.domain.Calculator
+import edu.nextstep.camp.calculator.domain.Expression
+import edu.nextstep.camp.calculator.domain.Operator
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class CalculatorViewModel(
+    initialExpression: Expression = Expression.EMPTY,
+    private val calculator: Calculator = Calculator()
+) : ViewModel() {
+
+    private val _expression = MutableStateFlow(initialExpression)
+    val expression = _expression.asStateFlow()
+
+    private val _incompleteExpressionErrorEvent = MutableSharedFlow<Unit>()
+    val incompleteExpressionErrorEvent = _incompleteExpressionErrorEvent.asSharedFlow()
+
+    fun addToExpression(operand: Int) = viewModelScope.launch {
+        _expression.emit(expression.value + operand)
+    }
+
+    fun addToExpression(operator: Operator) = viewModelScope.launch {
+        _expression.emit(expression.value + operator)
+    }
+
+    fun removeLast() = viewModelScope.launch {
+        _expression.emit(expression.value.removeLast())
+    }
+
+    fun calculate() = viewModelScope.launch {
+        val result = calculator.calculate(expression.value.toString())
+        if (result == null) {
+            _incompleteExpressionErrorEvent.emit(Unit)
+        } else {
+            _expression.emit(Expression(listOf(result)))
```

Comment:
```suggestion
            _expression.value = Expression(listOf(result))
```
StateFlow는 cold stream이므로, emit대신 value를 활용해보면 어떨까요?

Reply:
[17ca1e9](https://github.com/next-step/android-calculator-mvvm/pull/102/commits/17ca1e9da236aa8f4f8bc1a6c2bacd3936a211fc)